File: samples/snippets/fsharp/VS_Snippets_CLR_System/system.Random.Next/FS/next4.fs
open System

// <Snippet4>
let rnd = Random()

let malePetNames =
    [| "Rufus"; "Bear"; "Dakota"; "Fido";
        "Vanya"; "Samuel"; "Koani"; "Volodya";
        "Prince"; "Yiska" |]
let femalePetNames = 
    [| "Maggie"; "Penny"; "Saya"; "Princess";
        "Abby"; "Laila"; "Sadie"; "Olivia";
        "Starlight"; "Talla" |]

// Generate random indexes for pet names.
let mIndex = rnd.Next(0, malePetNames.Length)
let fIndex = rnd.Next(0, femalePetNames.Length)

// Display the result.
printfn "Suggested pet name of the day: "
printfn "   For a male:     %s" malePetNames.[mIndex]
printfn "   For a female:   %s" femalePetNames.[fIndex]

// The example displays output similar to the following:
//       Suggested pet name of the day:
//          For a male:     Koani
//          For a female:   Maggie
// </Snippet4>

File: snippets/csharp/System/Random/Overview/next4.cs
using System;

public class Example11
{
    public static void Main()
    {
        // <Snippet4>
        Random rnd = new();
        string[] malePetNames = [ "Rufus", "Bear", "Dakota", "Fido",
                                "Vanya", "Samuel", "Koani", "Volodya",
                                "Prince", "Yiska" ];
        string[] femalePetNames = [ "Maggie", "Penny", "Saya", "Princess",
                                  "Abby", "Laila", "Sadie", "Olivia",
                                  "Starlight", "Talla" ];

        // Generate random indexes for pet names.
        int mIndex = rnd.Next(0, malePetNames.Length);
        int fIndex = rnd.Next(0, femalePetNames.Length);

        // Display the result.
        Console.WriteLine("Suggested pet name of the day: ");
        Console.WriteLine($"   For a male:     {malePetNames[mIndex]}");
        Console.WriteLine($"   For a female:   {femalePetNames[fIndex]}");

        // The example displays the following output:
        //       Suggested pet name of the day:
        //          For a male:     Koani
        //          For a female:   Maggie
        // </Snippet4>
    }
}
